### PR TITLE
Add a .travis.yml file

### LINKED ISSUE #143: Investigate Travis CI
Possible to use Travis CI in replacement of Jenkins ? 
Although using the academic licence for private repo ?
Making everything public if necessary ?
Discuss with @dpshelio on the feasibility

### PR DESCRIPTION
**Pull request by UCLGuichard**
Fixes #143

### CODE REVIEW

File: cpp/purify/CMakeLists.txt [hunk 1]
```diff
@@ -25,17 +25,16 @@ set(HEADERS
...
   list(APPEND SOURCES casacore.cc)
   list(APPEND HEADERS casacore.h)
 endif()
 
 if(PURIFY_MPI)
-  list(APPEND HEADERS mpi_utilities.h distribute.h wavelet_operator_factory.h
-                      measurement_operator_factory.h operators.h convergence_factory.h)
-  list(APPEND SOURCES mpi_utilities.cc distribute.cc operators.cc )
+  list(APPEND HEADERS mpi_utilities.h distribute.h)
+  list(APPEND SOURCES mpi_utilities.cc distribute.cc)
```
Comment (dpshelio) [inline cpp/purify/CMakeLists.txt]: I'm confused... you've removed 
`wavelet_operator_factory.h`, `measurement_operator_factory.h`, `operators.h` and `convergence_factory.h` however only changed `mpi_utilities.h`  and `wavelet_operator_factory.h` - did you forget to push these changes?

File: cpp/purify/mpi_utilities.h [hunk 2]
```diff
@@ -3,10 +3,12 @@
 
 #include "purify/config.h"
 #include <vector>
-#include <sopt/mpi/communicator.h>
 #include <sopt/linear_transform.h>
 #include "purify/utilities.h"
 
+#ifdef PURIFY_MPI
+#include <sopt/mpi/communicator.h>
+#endif
```
Comment (dpshelio) [inline cpp/purify/mpi_utilities.h]: Why would `mpi_utilites` need guards if is `mpi_utilities`...??? Shouldn't be just used when used with mpi??

**Comment by dpshelio** [comment]
This is getting somewhere... slowly. For getting mpi working on the linux boxes I had to install from sources newer versions of openmpi and mpich (that's why these builds take ~30 min each). After lots of changing path battles (as you can see in the commits) I've managed to make it work, though openmpi builds complained that 4 nodes were too much.

There other problems to solve in the other builds:
- [ ] OSx builds with GCC. Somehow the tests are failing because the data is not seeing by the tests. Sometimes one test finds a file, but the following doesn't and removes it from memory. I don't really understand what's going on there. The clang build (no openMP, no mpi) works fine!!
- [ ] Linux with clang. It produces a seg-fault in one of the tests. I've not had time to dive on why.

@UCLGuichard, @Luke-Pratley would you be able to work in these? We could simply solve that by saying: Only GCC is supported on linux and Clang on OSx.

**Comment by UCLGuichard** [comment]
Hi @dpshelio sure ! I'm happy to do anything that could help in this direction.

**Comment by dpshelio** [comment]
Damn!! OpenMPI still fails. Now cmake finds it, but even with two processors keep failing... :cry:

**Comment by UCLGuichard** [comment]
argh :(

**Comment by Luke-Pratley** [comment]
@dpshelio This is really odd. I have been using gcc on OSX, with openMPI. Is it an issue of not finding the test files?

**Comment by UCLGuichard** [comment]
Hi @dpshelio and @Luke-Pratley , I am working on it now. Just quick remarks, it seems that only MPICH works for Linux. I am going to run a series of scaling tests to confirm that. I checked that on Legion I've been using Intel-MPI. However, there is OpenMPI there. Would it make sense to run the tests on Legion to see what happen ? I also have a VM for emulating Travis. Maybe worth checking what is going on there too.

**Comment by UCLGuichard** [comment]
Another point, does Clang allow OpenMP/MPI ? I see in the tests that you haven't set that option, is to worth checking ?

**Comment by UCLGuichard** [comment]
Note: I would suggest somewhere not to `make install` Sopt.

**Comment by UCLGuichard** [comment]
Note3: About Clang, could this be related ? 
https://github.com/UCL/STIR/issues/117 
At least, I've got Clang not finding `libiomp` on my VM.

**Comment by UCLGuichard** [comment]
Hi @Luke-Pratley and @dpshelio why is the `.travis.yaml` full of unresolved conflict markers ?

**Comment by UCLGuichard** [comment]
For the record, how to build using Clang, OpenMP with CMAKE:
```
cmake -DCMAKE_CXX_COMPILER=/usr/bin/clang++-5.0 -DCMAKE_C_COMPILER=/usr/bin/clang-5.0 -DOpenMP_CXX_FLAGS="-fopenmp=libiomp5" ..
```

**Pull request closed without merging**